### PR TITLE
Add documentation for Axes3d.view_init() angles

### LINKED ISSUE #28353: [MNT]: mplot3d azim-elev-roll order
### Summary

Part of PR #21426 'Add ability to roll the camera in 3D plots' (by Scott Shambaugh, 3 years ago) read as follows:
`- Make elev, azim, roll ordering consistent everywhere`

However, the choice of ordering is rather unfortunate:
- It is not the common one (most everyone else in the world uses azim-elev-roll, see below)
- It does not correspond to the order in which the 3 rotations take place (that's azim-elev-roll)
- It is not consistent with the ordering in matplotlib's `colors.py`.

I.e., there is room for improvement.

-----
 When I google:
- `"azimuth, elevation"` -> 291 000 results
- `"elevation, azimuth"`-> 41 500 results

That is to say, the `elevation, azimuth` ordering is not the common one. And if we consider that searching _for both_ yields:
- `"azimuth, elevation" and "elevation, azimuth"` -> 37 500 results (some links are non-committal), 

then it becomes clear that the `elevation, azimuth` ordering is really a minority position: (41-37)/(291-37) < 2%.
(Also, for what it's worth: MATLAB happens to use `"azimuth, elevation"`, just like (almost) everyone else.)

-----
The ordering with elevation first does not correspond to the order in which the 3 rotations take place:
if thinking about intrinsic rotations, then the rotation order is azim, elev, then roll.
The corresponding quaternion is
`    q = exp((+roll/2)*e_x)*exp((+elev/2)*e_y)*exp((-azim/2)*e_z)`
i.e., to be precise, mplot3d's `azim, elev, roll` are a kind of Tait-Bryan angles, -z,y',x".
It is a bit of a mess with the Euler angles, apparently there are 24 variations (not counting the signs) - https://github.com/moble/quaternion/wiki/Euler-angles-are-horrible

### Proposed fix

Switch (back) to the common ordering, the order in which the rotations are performed: `azim, elev, roll`.

Fortunately, Scott had the good foresight to use named parameters, so it is not a completely breaking change.
We can make the use of named parameters for `azim` and `elev` mandatory (preceding with a '*' argument), so changing the order (when parameters are unnamed) won't result in any silent failures.

Scott voiced a hesitation to change it (again), and I can understand his initial reaction. However, it seems to me that changing the order has clear benefits, and arguably it could be worth the effort of adapting existing code, as per https://matplotlib.org/devdocs/devel/api_changes.html (especially in view of the fact that the effort might be limited, with the named parameters).

I have a PR almost ready to show what the changes would be.

### PR DESCRIPTION
**Pull request by MischaMegens2**
## PR summary
This is an alternative for PR #28395, it adds documentation to clarify the relation between the Axes3d.view_init() angles and the rotation of the scene, and add recommendation to use keyword arguments; but unlike #28395, it leaves the original order (elev, azim, roll) unaffected. It closes Issue #28353.

## PR checklist

- [x] "closes #28353" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [x] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

File: lib/mpl_toolkits/mplot3d/axes3d.py [hunk 1]
```diff
@@ -1114,6 +1113,40 @@ def view_init(self, elev=None, azim=None, roll=None, vertical_axis="z",
         -YZ          0     180
         ==========   ====  ====
 
+        A roll angle of 0, 90, 180, or 270 degrees will rotate these views
+        while keeping the axes horizontal or vertical.
+
+        The *azim*, *elev*, *roll* angles correspond to rotations of the scene
```
Comment (timhoffm) [inline lib/mpl_toolkits/mplot3d/axes3d.py:1119]: A textual description is quite hard to follow. I suggest to instead link to https://matplotlib.org/devdocs/api/toolkits/mplot3d/view_angles.html#how-to-define-the-view-angle. And maybe move the quarternions there as well (in another?).

File: lib/mpl_toolkits/mplot3d/axes3d.py [hunk 2]
```diff
@@ -1606,13 +1639,8 @@ def _on_move(self, event):
...
             # update view
             vertical_axis = self._axis_names[self._vertical_axis]
-            self.view_init(
-                elev=elev,
-                azim=azim,
-                roll=roll,
-                vertical_axis=vertical_axis,
-                share=True,
-            )
+            self.view_init(elev, azim, roll, vertical_axis=vertical_axis,
```
Comment (timhoffm) [inline lib/mpl_toolkits/mplot3d/axes3d.py:1642]: Not that I care too much, but why are we going back here to positional arguments?

**Review by timhoffm — COMMENTED** [review]
Thanks for rewriting!

File: lib/mpl_toolkits/mplot3d/axes3d.py [hunk 3]
```diff
@@ -1114,6 +1113,40 @@ def view_init(self, elev=None, azim=None, roll=None, vertical_axis="z",
...
+        −z, +y', +x", rather than classic `Euler angles
+        <https://en.wikipedia.org/wiki/Euler_angles>`_.
+
+        To avoid confusion, provide the view angles as keyword
+        arguments: ``.view_init(elev=30, azim=-60, roll=0, ...)``.
+        This specific order is consistent with the order of positional arguments.
+        Unfortunately, this order does not match the order in which rotations are
+        applied, and it differs from the order used in other programs (``azim, elev``)
+        and in `matplotlib.colors.LightSource`. However, it cannot be
+        changed, since that would compromise backward compatibility.
```
Comment (scottshambaugh) [inline lib/mpl_toolkits/mplot3d/axes3d.py:1147]: I don't think this paragraph is needed, IMO it creates more confusion than it relieves.

File: lib/mpl_toolkits/mplot3d/axes3d.py [hunk 4]
```diff
@@ -1114,6 +1113,40 @@ def view_init(self, elev=None, azim=None, roll=None, vertical_axis="z",
...
+        The *azim*, *elev*, *roll* angles correspond to rotations of the scene
+        observed by a stationary camera, as follows (assuming a default vertical
+        axis of 'z'). First, a left-handed rotation about the z axis is applied
+        (*azim*), then a right-handed rotation about the (camera) y axis (*elev*),
+        then a right-handed rotation about the (camera) x axis (*roll*). Here,
+        the z, y, and x axis are fixed axes (not the axes that rotate together
+        with the original scene).
+
+        If you would like to make the connection with quaternions (because
+        `Euler angles are horrible
```
Comment (scottshambaugh) [inline lib/mpl_toolkits/mplot3d/axes3d.py:1128]: This is an excellent reference, but I think the tone should be more neural. E.g. "See X for a discussion on why quaternions are a more natural way to represent orientations)

Comment (MischaMegens2) [reply lib/mpl_toolkits/mplot3d/axes3d.py:1128]: It is being changed.

Comment (timhoffm) [reply lib/mpl_toolkits/mplot3d/axes3d.py:1128]: > It is being changed.

What do you mean by this? I don’t see any changes here. Did you forget to push?

Comment (MischaMegens2) [reply lib/mpl_toolkits/mplot3d/axes3d.py:1128]: Oh I'm sorry, it is less than clear, I meant to say: I'm changing it and I'll push a commit once I've tried to address the other issues raised too. (It may take a little while.)
